nrf802154_sniffer/nrf802154_sniffer.py

```diff
@@ -96,7 +96,10 @@ def stop_sig_handler(self, *args, **kwargs):
             self.running.clear()
             
             for thread in self.threads:
-                thread.join()
+                try:
+                    thread.join()
```

Comment:
I think there should be some timeout period.
```suggestion
                    thread.join(timeout=10)
                    if thread.is_alive() is True:
                        self.logger.error("Failed to stop a thread")
                        alive_threads.append(thread)  # New list object
                        # Also modify line 104 to self.threads = alive_threads
```